### DISCUSSION TITLE
Setting attributes annotated with `@Id` to null

### DISCUSSION THREAD

**Discussion by Spookyguy** [instancio/instancio#1183] [Q&A]
We are using Instancio to create test data in integration tests. When saving new entities to the database, the attribute of the entity annotated with @Id has to be null for the database to recognize it as a new entity. Currently we are setting the attribute to null manually like so:

```
Instancio.of(Entity.class).set(field(Entity::getId), null).create();
```

Is there a way to tell Instancio to set attributes annotated with @Id to null globally?

**Comment by armandino** [comment]
You can set it globally by registering a `InstancioServiceProvider` SPI implementation:

https://www.instancio.org/user-guide/#instancio-service-provider-interface

Here's a sample implementation:

```java
public class SampleInstancioServiceProvider implements InstancioServiceProvider {

    private static final Generator<String> NULL_GENERATOR = random -> null;

    @Override
    public GeneratorProvider getGeneratorProvider() {
        return (node, generators) -> {
            Field field = node.getField();

            if (field != null && field.getDeclaredAnnotation(Id.class) != null) {
                return NULL_GENERATOR;
            }

            return null;
        };
    }
}
```
